nemo/collections/nlp/models/information_retrieval/megatron_gpt_reranker_model.py

```diff
@@ -0,0 +1,317 @@
+                truncation_method=data_cfg.get(
+                    'truncation_method', 'right'
+                ),  # used to choose truncation method. Options: ['random', 'left', 'right']
+                special_tokens=self.cfg.data.get(
+                    'chat_prompt_tokens', None
+                ),  # special tokens for the chat prompts, a dictionary of {token_type: token}. Default: {'system_turn_start': '<extra_id_0>', 'turn_start': '<extra_id_1>', 'label_start': '<extra_id_2>', 'end_of_turn': '\n', "end_of_name": "\n"}
+                data_type="train" if is_train else "validation",
+            )
+            datasets.append(dataset)
+        if is_train:
+            if packed_sequence:
+                num_train_samples_after_blend = sum(len(dataset) for dataset in datasets)
+            dataset = BlendableDataset(
+                datasets=datasets, weights=data_cfg.concat_sampling_probabilities, size=num_train_samples_after_blend
+            )
+            return dataset
+        else:
+            return datasets
+
+    def training_step_fwd_bwd_step_call(self, dataloader_iter, forward_only):
+        loss_mean, non_loss_tensors = self.fwd_bwd_step(dataloader_iter, forward_only)
+        logit_diff = non_loss_tensors['logit_diff'][0].item()
+        self.log("logit_diff", logit_diff, prog_bar=True, rank_zero_only=True, batch_size=1)
+        return loss_mean
+
+    def inference_step_validation_call(self, batch, batch_idx, data_cfg, dataloader_idx=0):
+        metadata = batch.get('metadata', [{}] * len(batch['tokens']))
+        loss, non_loss_tensors = self.local_validation_step(itertools.chain([dataloader_idx], [batch]))
+        outputs = {
+            'loss': loss,
+            'metadata': metadata,  # [dict]
+            'query_pos_doc_logit': non_loss_tensors['query_pos_doc_logit'],  # [batch_size, hidden_size]
+        }
+        return outputs
+
+    def inference_loss_func(self, loss_mask, num_valid_tokens_in_ub, eos_tensors):
+        query_pos_doc_hs = eos_tensors
+        _blank = torch.zeros(1, device=query_pos_doc_hs.device, dtype=query_pos_doc_hs.dtype)[0]
+        return {
+            "loss": _blank,
+            "query_pos_doc_logit": query_pos_doc_hs,
+            "query_neg_doc_logit": _blank,
+            "logit_diff": _blank,
+        }
+
+    def _gather_global_inbatch_representations(self, local_eos_tensor):
+        local_eos_tensor = local_eos_tensor.contiguous()
+        global_eos_tensors = [
+            torch.zeros_like(local_eos_tensor) for _ in range(parallel_state.get_data_parallel_world_size())
+        ]
+        torch.distributed.all_gather(
+            global_eos_tensors, local_eos_tensor, group=parallel_state.get_data_parallel_group()
+        )
+        global_eos_tensors[parallel_state.get_data_parallel_rank()] = local_eos_tensor
+        global_eos_tensors = torch.cat(global_eos_tensors, dim=0)
+        return global_eos_tensors
```

Comment:
these two functions don't use "self", i wonder if it's better to make them helper functions outside of this class?